security/pkg/pki/util/utils.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// The key to specify corresponding service account in the annotation of K8s secrets.
+	ServiceAccountNameAnnotationKey = "istio.io/service-account.name"
+
+	// The ID/name for the certificate chain file.
+	CertChainID = "cert-chain.pem"
+	// The ID/name for the private key file.
+	PrivateKeyID = "key.pem"
+	// The ID/name for the CA root certificate file.
+	RootCertID = "root-cert.pem"
+	// cACertChainID is the CA certificate chain file.
+	cACertID = "ca-cert.pem"
+	// cAPrivateKeyID is the private key file of CA.
+	cAPrivateKeyID = "ca-key.pem"
+)
+
+func CreateSecret(saName, scrtName, namespace string, certChain, privateKey, rootCert, caCert, caPrivateKey []byte, secretType v1.SecretType) *v1.Secret {
```

Comment:
Public function needs doc string, including each parameter's usage.

Also I suggest to rename this to BuildSecret, since it only returns a go struct pointer, not adding actual resources in k8s.